### PR TITLE
Random token creation while testing.

### PR DESCRIPTION
**Pull request by isht3**

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import json
+import os
+import random
+import string
+
+from os.path import expanduser
+
+
+FILE = 'token.json'
+AUTH_TOKEN_PATH = "{}/.evalai/".format(expanduser('~'))
```
Comment (varunagrawal) [inline tests/base.py]: We need to make a list of things that need to go in a common config file.

File: tests/base.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class BaseTest:
+
+    @classmethod
+    def setup_class(cls):
+        if not os.path.exists(AUTH_TOKEN_PATH):
+            os.makedirs(AUTH_TOKEN_PATH)
+            with open(AUTH_TOKEN_PATH + FILE, 'w') as fw:
+                token = ''.join(random.choice(string.ascii_lowercase) for _ in range(40))
```
Comment (varunagrawal) [inline tests/base.py]: Use `random.choices(string.ascii_lowercase, k=40)` to make this line more terse and easier to read.
Also, you may want to set a seed for the PRNG in order to make this reproducible.

File: tests/base.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+AUTH_TOKEN_PATH = "{}/.evalai/".format(expanduser('~'))
+
+
+class BaseTest:
+
+    @classmethod
+    def setup_class(cls):
+        if not os.path.exists(AUTH_TOKEN_PATH):
+            os.makedirs(AUTH_TOKEN_PATH)
+            with open(AUTH_TOKEN_PATH + FILE, 'w') as fw:
```
Comment (varunagrawal) [inline tests/base.py]: Please use `os.path` to join paths. This ensures there are no issues on systems like Windows where the path delimiter is `\`.
E.g.
```python
import os.path as osp
with open(osp.join(AUTH_TOKEN_PATH, FILE), 'w') as fw:
```

Starting from Python 3.6, we have a new library called `pathlib` which I personally prefer, but we gotta be backwards compatible. :/

File: tests/base.py (hunk 2 above)
Comment (isht3) [reply tests/base.py]: Only thing is, `random.choices` is only for Python 3.6 and it's not backward compatible. So I should stick with this, yeah?

File: tests/base.py (hunk 1 above)
Comment (isht3) [reply tests/base.py]: Okay, I'll start an issue.

Comment (isht3) [reply tests/base.py]: @varunagrawal can you check #37, I've added this. :)

File: tests/base.py (hunk 2 above)
Comment (varunagrawal) [reply tests/base.py]: Ah I see. Good catch.

**Comment by isht3** [comment]
@varunagrawal This is done too!

**Comment by varunagrawal** [comment]
LGTM

**Comment by isht3** [comment]
@RishabhJain2018 Done!

**Comment by RishabhJain2018** [comment]
@isht3 @varunagrawal Since I have added these changes in #42 . Hence closing this.

**Pull request closed without merging**